lib/Dialect/TritonGPU/Transforms/AccelerateMatmul.cpp

```diff
@@ -655,27 +658,30 @@ class ScaledBlockedToMMAv5
       else if (dotOp.getBElemType() == ScaleDotElemType::E2M1)
         IsBMixedPrecFp4 = true;
     }
-
+    bool isMMAv5Fp4PaddedLhs = IsAMixedPrecFp4 || !dotOp.getLhsKPack();
+    bool isMMAv5Fp4PaddedRhs = IsBMixedPrecFp4 || !dotOp.getRhsKPack();
     // For mixed-precision fp4 operands, set allowTranspose = false, to force
     // the packed axis, K, to be contiguous in SMEM
     a = getSharedMemoryMMAOperand(a, rewriter, 0,
                                   /*allowTranspose=*/!IsAMixedPrecFp4,
-                                  IsAMixedPrecFp4, dotOp);
+                                  /*isMMAv5Fp4Padded=*/isMMAv5Fp4PaddedLhs,
+                                  /*forceTranspose=*/!dotOp.getLhsKPack(),
+                                  dotOp);
     b = getSharedMemoryMMAOperand(b, rewriter, 1,
                                   /*allowTranspose=*/!IsBMixedPrecFp4,
-                                  IsBMixedPrecFp4, dotOp);
+                                  /*isMMAv5Fp4Padded=*/isMMAv5Fp4PaddedRhs,
+                                  /*forceTranspose=*/!dotOp.getRhsKPack(),
+                                  dotOp);
 
     MLIRContext *context = dotOp->getContext();
     unsigned m = 128;
     unsigned n = retShapePerCTA[1] >= 256 ? 256 : retShapePerCTA[1];
     unsigned k = 32;
-    // If both operands are E2M1, target the FP4 tensor core implicitly.
-    // This may result in a downstream compile-time error if the scaled TC
-    // descriptor requires options that are unavailable to the .kind=mxf4 mma.
-    // This is likely preferable over a silent runtime performance degradation
-    // from running f4xf4 via .kind=mxf8f6f4
+    // If both operands are E2M1 and packed along K, target the FP4 tensor core.
+    // Otherwise target .kind=mxf8f6f4.
     if (dotOp.getAElemType() == ScaleDotElemType::E2M1 &&
-        dotOp.getBElemType() == ScaleDotElemType::E2M1) {
+        dotOp.getBElemType() == ScaleDotElemType::E2M1 && dotOp.getLhsKPack() &&
+        dotOp.getRhsKPack()) {
       k = 64;
     }
```

Comment:
Can we change this condition to allow for an early perf warning like the below?

```suggestion
      dotOp.getBElemType() == ScaleDotElemType::E2M1) {
      if (dotOp.getLhsKPack() || dotOp.getRhsKPack()) {
        dotOp->emitWarning() << "Warning: falling back to FP8 tensor core because K-major operand requirements for native FP4 are not met\n" << "* LhsKPack = " << dotOp.getLhsKPack() << "\n* RhsKPack = " << dotOp.getRhsKPack();
        k = 32;
      } else {
        k = 64;
      }
    }
```

Reply:
I'm a bit reluctant on adding a perf warning here as user most likely needs the transposition behavior if they are in this scenario and having a warning will be noisy. The default behavior of dot_scaled is to pack along K which should ideally get users to use that when possible.
We should probably document the performance behavior somewhere although I don't have a good suggestion right now.

Reply:
My lean is usually toward being noisy when we know a significant performance degradation is triggered. Otoh, documenting that using non-default values for the lhs/rhskpack fields in the op as potentially impacting performance works too. I don't feel great about not letting the user know about this contract at all, but either of these approaches are good to me

Reply:
also turns out this code was dead, so removed it altogether